### ISSUE TITLE
Multisite support

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#1029]
We should support [WordPress Multisite](http://codex.wordpress.org/Create_A_Network).

The main question is how to physically split the sites in the network. Should they have separate Git repos each? Should there be a central repo for network-wide actions? How will workflows with staging and similar work? Etc.

Creating this ticket to gather feedback and ideas.

**Comment by cjmaio** [comment]
I think that a central repo for network-wide actions, and then individual repos for each site would be beneficial - in cases where you might want to use the Git repo to restore a multisite site to a regular WordPress install.

**Comment by borekb** [comment]
Splitting/merging a site from/to multisite is an intriguing feature, I agree.

**Comment by kalich5** [comment]
Multisite support is important. And with the described functions also agree.

**Comment by mthchz** [comment]
I think that a central repo in the network admin is the better.
For me, WP Multisite is 1 network = 1 database = 1 repo.

Good luck,
hope the feature come soon :)

**Comment by seagyn** [comment]
I agree with @mthchz. It also may role support out for it more quickly to have 1 repo. Worry about the per site feature later.

**Comment by wademac** [comment]
I would love to see this do multisites :)

**Comment by fabiobfava** [comment]
I wish we could rely on it for pushing **from** _Live_ to _Staging_ or _Dev_ (_Local_). Not necessairily to reverse, since for me the most important is a reliable _Multi Network Support_, mostly to be able to _test for updates_ on those environments before updating the _Live Sites_.

**Comment by americloud** [comment]
I need this as well. 

For me its 1 network = 1 DB = 1 repo

Any changes made in the entire WP instance should be version controlled. 

Would love to see this very soon! :)

**Comment by amjad** [comment]
I need this as well.

For me its 1 network = 1 DB = 1 repo

Any changes made in the entire WP instance should be version controlled.

Would love to see this very soon! :)

**Comment by pentatonicfunk** [comment]
my suggestion will be probably hated by everyone
But how about we use central git repo as site-network actions, and another git-repos for each subsite but embedded as submodule or this one <https://github.com/ingydotnet/git-subrepo#readme> pretty interesting

**Comment by amjad** [comment]
When will that one day come? This is badly needed, I’m sick of using GoodSync.

**Comment by borekb** [comment]
@amjad Realistically speaking, multisite support will not come very soon :/ First, we need to ship VersionPress 4.0 which is quite a bit of work in its own right. Then we'd like to do 5.0 with features that have been neglected in 4.0 (where the focus was almost exclusively on plugin support); there might also be things like Gutenberg compatibility. Only after all that, we can start thinking about multisite support.

Of course, this is our priorities. If you strongly need multisite support and can implement it yourself or hire some engineers to do it for you, we'd be more than happy to merge it sooner. That is always an option.

**Comment by amjad** [comment]
bummer. So currently if the plugin is installed on the main site of a multi-site install, and since the code base is the same for all multi-site instances, will the changes to any files show up here? I am not very concerned about the database changes at this time, I can rely on the full backups for now. I only care about having source control on my WP files. Will this plugin still work in my case? I'd like to know before I try.

**Comment by borekb** [comment]
@amjad For files only, you don't need VersionPress; plain Git is more than enough.

**Comment by juslintek** [comment]
Tell me what is stopping from making this work, maybe I can fork or you can make a branch for me. I really need this feature.

**Comment by borekb** [comment]
@juslintek Awesome that you're willing to help with this! Feel free to fork the repo and send a pull request (or more :) ), here are our contributing guidelines: [`CONTRIBUTING.md`](https://github.com/versionpress/versionpress/blob/master/CONTRIBUTING.md).

**Comment by juslintek** [comment]
@camilleterol sorry, not yes was so busy on my work and freelancing that I could spare any time on this project currently, but maybe you if you have any free time and have nowhere else to put it could start with it. Then hopefully imbetween Christmas and New Years holidays, I'll be off my full time job, maybe I'll be able to look into it.

**Comment by kalich5** [comment]
??

**Comment by dsnid3r** [comment]
I'd like to see this picked back up...Any way I can help?

**Comment by borekb** [comment]
For anyone who would like to see this happen and is willing to help, it would be awesome to read a detailed proposal of this feature – how would it work, which actions would lead to which commits, how would the data structures be tracked, if you'd like to see a single repo model vs. multiple related repo (and in case of multi repo, how would you visualize that in VersionPress UI), etc.

Multisite support is still not our immediate priority but in the meantime, it would be good to gather as much information as possible. This will be a relatively hard endeavor so any details will greatly help. Thank you!

**Comment by dsnid3r** [comment]
@borekb I might have one of my developers I can pass this onto for the actual development of it (I have very limited time but will help however I can). This is a feature I've been searching for, it would be best like you said borekb if we frameworked this out of how it works so whoever the dev is knows exactly what we're looking for.

**Comment by amjad** [comment]
I mentioned my workflow and requirements in my post above. Still waiting for multisite support.

**Comment by borekb** [comment]
@dsnid3r Contributing development time would be absolutely awesome!

**Comment by dsnid3r** [comment]
@borekb Its time to get this going one way or another. Do we have a direction nailed down on how we want this to function from the splitting sense? IMO I'd really like to see a staging branch and production branch aspect with the ability to deploy and rollback the subsites across multiple instances as desired. What specific issues does VersionPress currently have from running the entire network as one single repository, possibly to start?

**Comment by borekb** [comment]
@dsnid3r Hi Dustin, we don't actively work on VersionPress at the moment, see #1481, but if you want to work out exactly how Multisite would work or possibly even implement it, certainly go ahead.
